src/platform/Tizen/WiFiManager.h

```diff
@@ -0,0 +1,93 @@
+public:
+    static void Init(void);
+    static void Deinit(void);
+
+    static CHIP_ERROR IsActivated(bool * isWifiActivated);
+    static CHIP_ERROR Activate(void);
+    static CHIP_ERROR Deactivate(void);
+    static CHIP_ERROR Connect(const char * ssid, const char * key);
+    static CHIP_ERROR Disconnect(const char * ssid);
+    static CHIP_ERROR RemoveAllConfigs(void);
+
+    static CHIP_ERROR GetDeviceState(wifi_manager_device_state_e * deviceState);
+    static CHIP_ERROR SetDeviceState(wifi_manager_device_state_e deviceState);
+    static CHIP_ERROR GetModuleState(wifi_manager_module_state_e * moduleState);
+    static CHIP_ERROR GetConnectionState(wifi_manager_connection_state_e * connectionState);
+
+private:
+    static void _DeviceStateChangedCb(wifi_manager_device_state_e deviceState, void * userData);
+    static void _ModuleStateChangedCb(wifi_manager_module_state_e moduleState, void * userData);
+    static void _ConnectionStateChangedCb(wifi_manager_connection_state_e connectionState, wifi_manager_ap_h ap, void * userData);
+    static void _ScanStateChangedCb(wifi_manager_scan_state_e scanState, void * userData);
+    static void _RssiLevelChangedCb(wifi_manager_rssi_level_e rssiLevel, void * userData);
+    static void _BackgroundScanCb(wifi_manager_error_e wifiErr, void * userData);
+    static void _IPConflictCb(char * mac, wifi_manager_ip_conflict_state_e ipConflictState, void * userData);
+    static void _ActivateCb(wifi_manager_error_e wifiErr, void * userData);
+    static void _DeactivateCb(wifi_manager_error_e wifiErr, void * userData);
+    static void _ScanFinishedCb(wifi_manager_error_e wifiErr, void * userData);
+    static bool _FoundAPCb(wifi_manager_ap_h ap, void * userData);
+    static void _ConnectedCb(wifi_manager_error_e wifiErr, void * userData);
+    static bool _ConfigListCb(const wifi_manager_config_h config, void * userData);
+
+    static gboolean _WiFiInitialize(gpointer userData);
+    static gboolean _WiFiActivate(GMainLoop * mainLoop, gpointer userData);
+    static gboolean _WiFiDeactivate(GMainLoop * mainLoop, gpointer userData);
+    static gboolean _WiFiScan(GMainLoop * mainLoop, gpointer userData);
+    static gboolean _WiFiConnect(GMainLoop * mainLoop, gpointer userData);
+
+    void _WiFiDeinitialize(void);
+    void _WiFiSetStates(void);
+    void _WiFiSetCallbacks(void);
+    void _WiFiUnsetCallbacks(void);
+    void _WiFiSetDeviceState(wifi_manager_device_state_e deviceState);
+    void _WiFiSetModuleState(wifi_manager_module_state_e moduleState);
+    void _WiFiSetConnectionState(wifi_manager_connection_state_e connectionState);
+    wifi_manager_ap_h _WiFiGetFoundAP(void);
+
+    static WiFiManager sInstance;
+
+    wifi_manager_h mWiFiManagerHandle;
+    wifi_manager_device_state_e mDeviceState;
+    wifi_manager_module_state_e mModuleState;
+    wifi_manager_connection_state_e mConnectionState;
+
+    char mWiFiSSID[kMaxWiFiSSIDLength + 1];
+    char mWiFiKey[kMaxWiFiKeyLength];
```

Comment:
Why not the +1 here as well for null-terminator?

Also, maintenance of keys in memory (when they are only ever used seldom) is a bit risky. Consider letting the caller safely store the key, and do no persistently maintain it here unless truly necessary.

Reply:
DeviceNetworkInfo class is as follows.
```
    char WiFiSSID[kMaxWiFiSSIDLength + 1]; /**< The WiFi SSID as a NULL-terminated string. */
    uint8_t WiFiKey[kMaxWiFiKeyLength];    /**< The WiFi key (NOT NULL-terminated). */
```
And i referenced mSSID and mCredentials parameters used in DoEnableNetwork and ProvisionWiFi functions.
struct WiFiNetworkInfo of NetworkInfo is as belows.
```
    uint8_t mSSID[kMaxWiFiSSIDLen + 1];
    uint8_t mSSIDLen;
    uint8_t mCredentials[kMaxWiFiCredentialsLen];
    uint8_t mCredentialsLen;
```

Reply:
Yes, but DeviceNetworkInfo uses a length-delimited key and this patch uses a null-terminated key.  This patch also is using g_strlcpy to write mWiFiKey, and that will write a null character.

This buffer either needs to have an additional byte for the null character or we need to switch to a length-delimited key as DeviceNetworkInfo uses.

Yes, but code in this patch uses g_strlcpy to write to mWiFiKey and

Reply:
Thank you for your feedback.
As you said, if strlcpy is used, it should be + 1.
Will update.